### PR TITLE
comment usage bugfix

### PR DESCRIPTION
**Pull request by eugen-guenther-sap**

### CODE REVIEW

**Comment by lucasborin** [comment]
Would it be possible to enhance the unit test with the new scenario?

**Comment by larshp** [comment]
abaplint, note that it was only the last parenthesis that caused problems

just like in
`result = xsdbool( ( 1 = 2 ) OR 'bar' = 'foo' ).`
the parenthesis surrounds a comparison is not needed

image

**Comment by eugen-guenther-sap** [comment]
@lucasborin: It should be ok now.

@larshp: i fixed it 5 hours ago.

File: src/checks/y_check_comment_usage.clas.abap [hunk 1]
```diff
@@ -70,19 +77,25 @@ CLASS y_check_comment_usage IMPLEMENTATION.
...
+  METHOD is_comment_excluded.
+
+      DATA(token_len) = strlen( token ).
+      result = xsdbool( token+0(minimal_comment_len) = |*"| OR
+                        token+0(minimal_comment_len) = |"!| OR
+                        token+0(minimal_comment_len) = |##| OR
+                        token+0(minimal_comment_len) = |*?| OR
+                        token+0(minimal_comment_len) = |"?| OR
+                        token CO |*| OR
+                        token_len >= exception_comment_len AND token+0(exception_comment_len) = |"#E| OR
```
Comment (lucasborin) [inline src/checks/y_check_comment_usage.clas.abap:95]: The parenthesis is needed here. Isn't it?

File: src/checks/y_check_comment_usage.clas.abap [hunk 2]
```diff
@@ -70,19 +77,25 @@ CLASS y_check_comment_usage IMPLEMENTATION.
...
+                        token CP '"' && object_name && '*.'  OR
+                        token_len >= include_comment_len AND token+0(include_comment_len) = |* INCLUDE| ).
```
Comment (lucasborin) [inline src/checks/y_check_comment_usage.clas.abap:97]: The parenthesis is needed here. Isn't it?

File: src/checks/y_check_comment_usage.clas.abap (hunk 1 above)
Comment (eugen-guenther-sap) [reply src/checks/y_check_comment_usage.clas.abap:95]: if you like the magic numbers we can remove the anti-magic version

File: src/checks/y_check_comment_usage.clas.abap (hunk 2 above)
Comment (eugen-guenther-sap) [reply src/checks/y_check_comment_usage.clas.abap:97]: we only want the first 2,3,9 characters of the string.

**Review by lucasborin — APPROVED** [review]

**Pull request merged**